db.go

```diff
@@ -106,6 +111,33 @@ type Db interface {
 	// the database yet.
 	NewestSha() (sha *btcwire.ShaHash, height int64, err error)
 
+	// FetchAddrIndexTip returns the hash and block height of the most recent
+	// block which has had its address index populated. It will return
+	// ErrAddrIndexDoesNotExist along with a zero hash, and -1 if the
+	// addrindex hasn't yet been built up.
+	FetchAddrIndexTip() (sha *btcwire.ShaHash, height int64, err error)
+
+	// UpdateAddrIndexForBlock updates the stored addrindex with passed
+	// index information for a particular block height. Additionally, it
+	// will update the stored meta-data related to the curent tip of the
+	// addr index. These two operations are performed in an atomic
+	// transaction which is commited before the function returns.
+	// Addresses are indexed by the raw bytes of their base58 decoded
+	// hash160.
+	UpdateAddrIndexForBlock(blkSha *btcwire.ShaHash, height int64,
+		addrIndex *BlockAddrIndex) error
+
+	// FetchBlockBySha looks up and returns all transactions which either
+	// spend from a previously created output of the passed address, or
+	// create a new output locked to the passed address. The, `limit` parameter
+	// should be the max number of transactions to be returned. Additionally, if the
```

Comment:
Added a line to the documentation stating the parameters MUST be non-negative. Violation is handled by returning with an error.